### PR TITLE
Feature/187

### PR DESCRIPTION
**Pull request by delamux**
Update versions and main readme Spanish translation

### CODE REVIEW

File: hooks/14_FormValidation/src/pages/loginPage.tsx [hunk 1]
```diff
@@ -29,9 +29,9 @@ const useStyles = makeStyles(theme =>
   })
 );
 
-interface Props extends RouteComponentProps {}
+interface Props {}
 
-const LoginPageInner = (props: Props) => {
+export const LoginPage = (props: Props) => {
```
Comment (brauliodiez) [inline hooks/14_FormValidation/src/pages/loginPage.tsx:34]: Should we use here:

```tsx
export const LoginPage : React.FC<Props> = (props) => {
```

**Comment by brauliodiez** [comment]
Great stuff, Just found something to get updated on all function components (bad stuff introduced years ago :)), I think we should type all functional components, e.g.:

```diff
- export const HelloComponent = (props: Props) => (
+ export const HelloComponent  : React.FC<Props> = (props) => (
```

**Comment by brauliodiez** [comment]
I'm not sure if it would be a good idea to apply destructuring to all props in components, what do you think?

e.g.

```diff
interface Props {
  initialUserName: string;
  onNameUpdated: (newName: string) => any;
}

export const NameEditComponent = (props: Props) => {
+ const {initialUserName, onNameUpdated} = props;
  const [editingName, setEditingName] = React.useState(props.initialUserName);

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setEditingName(e.target.value);
  };

  const onNameSubmit = (event: any): any => {
-    props.onNameUpdated(editingName);
+    onNameUpdated(editingName);
  };

```

**Comment by brauliodiez** [comment]
Example 14_FormValidation should be migrated from lc-form-validation to Fonk + Formik, give a me a buzz if you need some guidance or if you want me to implement it

**Comment by delamux** [comment]
> I'm not sure if it would be a good idea to apply destructuring to all props in components, what do you think?
> 
> e.g.
> 
> ```diff
> interface Props {
>   initialUserName: string;
>   onNameUpdated: (newName: string) => any;
> }
> 
> export const NameEditComponent = (props: Props) => {
> + const {initialUserName, onNameUpdated} = props;
>   const [editingName, setEditingName] = React.useState(props.initialUserName);
> 
>   const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
>     setEditingName(e.target.value);
>   };
> 
>   const onNameSubmit = (event: any): any => {
> -    props.onNameUpdated(editingName);
> +    onNameUpdated(editingName);
>   };
> ```

**Comment by delamux** [comment]
> I'm not sure if it would be a good idea to apply destructuring to all props in components, what do you think?
> 
> e.g.
> 
> ```diff
> interface Props {
>   initialUserName: string;
>   onNameUpdated: (newName: string) => any;
> }
> 
> export const NameEditComponent = (props: Props) => {
> + const {initialUserName, onNameUpdated} = props;
>   const [editingName, setEditingName] = React.useState(props.initialUserName);
> 
>   const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
>     setEditingName(e.target.value);
>   };
> 
>   const onNameSubmit = (event: any): any => {
> -    props.onNameUpdated(editingName);
> +    onNameUpdated(editingName);
>   };
> ```

I don't know, in this case maybe is better call the props because is more readable and we know that comes from the props?

**Pull request closed without merging**